### ISSUE TITLE
Bad UI with blockades and pincer manoeuvres

### ISSUE THREAD

**Issue by MichaelCourtney** [freeorion/freeorion#4459]
23rd MP turn 114-115, post 75: https://freeorion.org/forum/viewtopic.php?t=12706&sid=4fc4b9ef2496c99b379d19b96b9a6788&start=75

If fleets coming from 2 directions hit a blockade, survive and are blockaded, the UI tells you:

![UI](https://user-images.githubusercontent.com/43386669/227610755-8315bbdd-8d54-4814-bcf1-f2c457974edf.png)

reality is the individual fleets can only return to the system they came from, not both systems.

If you now combine the ships into 1 fleet the UI will now allow you to select either system as a destination, however neither destination is possible as the new fleet contains ships unable to travel to system A, and ships unable to travel to system B. If you hit end turn the fleet will not move.

**Comment by Oberlus** [comment]
~~Forbidding to merge blockaded fleets is a possible solution.~~

**Comment by Blue-Award** [comment]
So actually the UI already is forbidding the merge, it seems, but it still shows individual fleets as safe to retreat in either direction, which can get confusing when you do pincer manouvers with multiple fleets and ships

**Comment by Vezzra** [comment]
So this is basically an UI issue about misleading information presented to the player, correct?

Do you guys really think this is serious enough that it needs to be fixed for 0.5? I've no problem with that, just want to make sure.

**Comment by geoffthemedio** [comment]
> Do you guys really think this is serious enough that it needs to be fixed for 0.5?

no

**Comment by MichaelCourtney** [comment]
> So this is basically an UI issue about misleading information presented to the player, correct?
> 
> Do you guys really think this is serious enough that it needs to be fixed for 0.5? I've no problem with that, just want to make sure.

I wouldn't delay release over it, at the end of the day its probably been there in every other release too.

**Comment by agrrr3** [comment]
The UI fix here would be not to allow the wrong fleet orders. 

Besides that a "Sort ships into fleets by obstruction" option would make probably sense.
Or change the way that obstruction works? Maybe a topic for the next developer meetup?

**Comment by Oberlus** [comment]
> The UI fix here would be not to allow the wrong fleet orders.

You mean the merging of fleets that can't leave through the same lane, right?

**Comment by agrrr3** [comment]
> > The UI fix here would be not to allow the wrong fleet orders.
> 
> You mean the merging of fleets that can't leave through the same lane, right?

Not necessarily. What is important is that the obstruction info is not lost when reordering ships in fleets. And that the UI supports it.

I am not sure if the arrival lane is not also recorded in the ship.
If it is only recorded in the fleet, yes the easiest way to prevent loosing obstruction info would be to disallow merging of fleets (disallow adding ships to a fleet) with  different obstruction info. 

If obstruction is recorded in each ship, better visualisation would suffice:
* A way to see/sort ships by their obstruction
* Marking obstruction in fleet (and maybe ships) and marking mixed obstruction in a fleet
* Being stopped from impossible orders (which was the main problem of the OP)

**Comment by agrrr3** [comment]
Probably you are right. With the current obstruction model it does not make sense to mix obstructed ships with different arrival lanes. The most important reason for merging fleets is synching arrival times. In this case the fleet can't move at all.

OTOH it can make sense to merge unobstructed ships into  obstructed fleets in order to flee synched.

**Comment by o01eg** [comment]
Saves for 114 and 115 turn, and logs: 
[FreeOrion_23th.zip](https://github.com/freeorion/freeorion/files/11477394/FreeOrion_23th.zip)
